### DISCUSSION TITLE
Lots of Bugs.. the app freezes hangs... (UPDATE FASTER ON RELEASE MODE)

### DISCUSSION THREAD

**Discussion by nextcodelab** [dotnet/maui#11344] [General]
(Android Only)
Its seems that flutter is lightweight and .Net Maui is heavyweight,,,
Can you please why flutter is super fast than Maui? 
How flutter render so fast on screen?
 even start-up super fast in flutter... 
We know that Maui is real native... but why flutter is super light and fast and you cannot feel the buggy screen in flutter?
Is it because the XAML Compile at runtime?
Maybe .net maui should compile xaml completely to real code without XAML footprints in release mode ?

The buggy screen only appear on android not on Windows and maybe not on iOS.
.NET MAUI IS FAST ON OTHER PLATFORMS... BUT ON ANDROID VERY LAG... I hope lag only  appear on debug mode.

This how smooth flutter is... no lags even in animations

https://user-images.githubusercontent.com/73583588/202335885-254194ae-9755-4a41-951d-ba639c231226.mp4

.Net Maui you can see unrenponsive clicking..

https://user-images.githubusercontent.com/73583588/202353386-bfaa7485-d090-43a8-a1d8-7c318d488a49.mp4

**Comment by manishbisen** [comment]
Interested to get answer on it.

**Comment by akhanalcs** [comment]
Hoping to see an answer.

**Comment by Mhamad-Raad** [comment]
Hi there
maybe this answers some of your questions when it comes to Flutter vs Maui: [article link](https://medium.com/pragmatic-programmers/google-flutter-vs-microsoft-net-maui-which-one-will-i-use-3015ec3f99a5#:~:text=Looking%20strictly%20at%20supported%20client,with%20perhaps%20slightly%20better%20performance).

and see if you can improve your app by following this discussion: [discussion link](https://github.com/dotnet/maui/issues/9485).

**Reply by akhanalcs** [reply]
The author of the linked article doesn't seem knowledgeable enough to comment on the topic.

**Comment by jamesmcmenamin** [comment]
Questions like this without specific code driven examples are nothing but derivative dribble.

The vast majority of people I’ve seen complaining over performance issues. From both Maui and from Xamarin. Do things like 

1) use bad practices in db’s
2) I have a whole bunch of crap going on the main thread of the app CS.
3) Think that it’s OK to have 400 lines of code in a code behind
4) zero ioc or any level of soc
5) overall just miss all the basics of object oriented programming

While constructive criticism is welcome and most people appreciate constructive criticism and poignant questions. This not CNN / Fox News / MSNBC. Leave the inflammatory statements there. Please provide some code examples with data driven testing / benchmarks when talking performance.

**Reply by nextcodelab** [reply]
This is how smooth flutter... I don't know why even this is not native, the  performance no lags...
No lag even in animations...

If we create .Net Maui app and like this... we can see and feel the difference of the performance..

https://user-images.githubusercontent.com/73583588/202336125-19864fa9-7003-4f4c-ba04-33e270bb41ff.mp4

**Reply by manishbisen** [reply]
@nextcodelab can you please post some similar videos of maui where performance is lag.So that difference can be seen clearly ?

**Reply by nextcodelab** [reply]
.Net Maui,,, there are some delays and unresponsiveness in clicking... 

https://user-images.githubusercontent.com/73583588/202353576-04fabd2d-a3b1-454c-8476-a6af3e3aaaa6.mp4

**Comment by nextcodelab** [comment]
It seems that the cause of bulkiness,  and feel the heavy or not responsive in .Net Maui,,, maybe becuase of the XAML compilation... it seems that XAML compile 50x before it landed to native view... I hope someday it can be improve in compilation.

YES it is real native BUT... Heavier xaml markup cause sometimes unresponsiveness, and we cannot design better UI if we do not write more XAML markup... so maybe I hope someday they improve the XAML Compilation.

THIS HAS POTENTIAL
https://github.com/jsuarezruiz/AlohaKit

**Comment by coderbasket** [comment]
Update Maui Android.
Release mode is faster compare to debug 5 seconds, vs release mode 2 seconds only... 
Maui is better now.
.NET MAUI START-UP TEST IN RELEASE MODE;

https://user-images.githubusercontent.com/97447955/203037492-4c19b617-90b6-48ba-948d-bfeb4b8586bf.mp4

**Reply by manishbisen** [reply]
@coderbasket Its not about hello world app,it's about real world apps.
